### ISSUE TITLE
[Improvement]: Move model changer to self

### ISSUE THREAD

**Issue by tiCkuhAr** [YimMenu/YimMenu#2236]
### Which feature

It would be nice if the model changer could be moved to the self tab to make it more user-friendly. Also, when changing models, it takes away all the user's guns, and you have to end up giving yourself guns again. Making it grab the user's current weapons and give them back after changing models would be nice. Persist weapons might be able to be used for this, making it less work to add

### Reason

The current model changer is really confusing, as it's in spawn ped, which does not make a lot of sense if I'm trying to change my player model. The last place I'm going to look is in spawn ped, or even in the world, for that matter.

**Comment by ghost** [comment]
![image](https://github.com/YimMenu/YimMenu/assets/20650873/02def73f-54b7-41f6-8b15-dc68b6cc8572)

case closed

**Comment by tiCkuhAr** [comment]
> ![image](https://user-images.githubusercontent.com/20650873/273502290-02def73f-54b7-41f6-8b15-dc68b6cc8572.png)
> 
> case closed

I take it you did not read the request at all lol

**Comment by ghost** [comment]
> > <snip>
> I take it you did not read the request at all lol

I did but I don’t get it. It seems simple to me.

**Comment by tiCkuhAr** [comment]
> > > 
> > 
> > 
> > I take it you did not read the request at all lol
> 
> I did but I don’t get it. It seems simple to me.

So in your mind, spawn ped = model changer cuss This request is aimed at making the menu more simple for new users. I already know where it is and how to use it. If you tried reading the request, you would have known that.

**Comment by ghost** [comment]
Ok ok fine. How about. There’s an actual model changer tab in self so it’d be Self -> Player Model and the ped spawner can just remove the option to change your player model. Okay, that makes sense. I guess you have a point making it easier for new users but if you already know where it is, it’s simple.

**Comment by tiCkuhAr** [comment]
> Ok ok fine. How about. There’s an actual model changer tab in self so it’d be Self -> Player Model and the ped spawner can just remove the option to change your player model. Okay, that makes sense. I guess you have a point making it easier for new users but if you already know where it is, it’s simple.

That's true. Using it is simple, but this request is not about using it; it's about finding it.

"The current model changer is really confusing, as it's in spawn ped, which does not make a lot of sense if I'm trying to change my player model. The last place I'm going to look is in spawn ped, or even in the world, for that matter."

**Comment by pierrelasse** [comment]
So final request would be:
- Move "Spawn Ped" to Self tab
- Try to make it keep the weapons when the model is switched? (This would require saving the weapons and loading them afterwards i think)

Correct?

**Comment by tiCkuhAr** [comment]
> So final request would be:
> 
>     * Move "Spawn Ped" to Self tab
> 
>     * Try to make it keep the weapons when the model is switched? (This would require saving the weapons and loading them afterwards i think)
> 
> 
> Correct?

Correct

**Comment by ghost** [comment]
Keeping weapons when changing ped sounds neat but beware of some animals that cause errors if you try to pull out a weapon. For example. the rabbit_02 model will throw errors and throw many errors, although it doesn't crash your game it's still pretty bad. Anyway, models that cause errors like that should probably be put on some sort of blacklist for giving weapons to you. 

Or simply just put a warning specifying that some animal models cause issues with weapons, and make the keep weapons thing a checkbox users can decide if they want or not.

**Comment by tiCkuhAr** [comment]
> Keeping weapons when changing ped sounds neat but beware of some animals that cause errors if you try to pull out a weapon. For example. the rabbit_02 model will throw errors and throw many errors, although it doesn't crash your game it's still pretty bad. Anyway, models that cause errors like that should probably be put on some sort of blacklist for giving weapons to you.
> 
> Or simply just put a warning specifying that some animal models cause issues with weapons, and make the keep weapons thing a checkbox users can decide if they want or not.

That might be something to make a bug report about.

**Comment by ghost** [comment]
> > Keeping weapons when changing ped sounds neat but beware of some animals that cause errors if you try to pull out a weapon. For example. the rabbit_02 model will throw errors and throw many errors, although it doesn't crash your game it's still pretty bad. Anyway, models that cause errors like that should probably be put on some sort of blacklist for giving weapons to you.
> 
> > 
> 
> > Or simply just put a warning specifying that some animal models cause issues with weapons, and make the keep weapons thing a checkbox users can decide if they want or not.
> 
> 
> 
> That might be something to make a bug report about.

Personally I don’t think it’s a bug at all. I think it’s just because the game doesn’t ever intend for you to have weapons as an animal. It’s always been that way, plus I don’t think it’s really a problem.

**Comment by pierrelasse** [comment]
Its a problem on gtas side which is not a problem bcause it is not intended to used with weapons.

**Comment by ghost** [comment]
> Its a problem on gtas side which is not a problem bcause it is not intended to used with weapons.

ik

**Comment by tiCkuhAr** [comment]
> Its a problem on gtas side which is not a problem bcause it is not intended to used with weapons.

I'm not sure I get your point. Most of the crash protections are all for GTA bugs.
